### PR TITLE
Fix reverse amount calculation for trades. When maxDestAmount is small and we recalculate trade values in reverse.

### PR DESCRIPTION
**Pull request by manhlx3006**
When doing reverse calculation:
- new src amounts are used only when any of them isn't greater then corresponding current srcAmount.
- If newTradeWei > tradeWei, fallback to use tradeWei.

### CODE REVIEW

File: contracts/sol6/KyberNetwork.sol [hunk 1]
```diff
@@ -1133,13 +1134,14 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
         }
 
         // reverse calculation, because we are working backwards
-        tradeData.tradeWei =
+        uint256 newTradeWei =
             (weiAfterFees * BPS * BPS) /
```
Comment (ilanDoron) [inline contracts/sol6/KyberNetwork.sol:1138]: maybe we should still improve calculation here.
like mul with BPS ** 4

anyway its better to mul with constant value. and avoid const * const
seems the compiler doesn't do it in compilation time...

Comment (Anyhowclick) [reply contracts/sol6/KyberNetwork.sol:1138]: My concern is that multiplying with that large a number might cause overflow problems

Comment (Anyhowclick) [reply contracts/sol6/KyberNetwork.sol:1138]: I guess it depends what's the maximum theoretical value of `calcTradeSrcAmount`?

Comment (ilanDoron) [reply contracts/sol6/KyberNetwork.sol:1138]: OK
seems to me we should not update this value.
worst case is a few wei error.
we can keep existing numbers.

File: contracts/sol6/KyberNetwork.sol [hunk 2]
```diff
@@ -1140,37 +1143,47 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
...
                 destAmountSplit,
                 srcDecimals,
                 destDecimals,
                 reservesData.rates[i]
             );
-            newSrcAmount = minOf(newSrcAmount, currentSrcAmount);
+            if (newSrcAmounts[i] > currentSrcAmount) {
+                // revert back to use current src amounts
+                newSrcAmount = srcAmount;
+                return newSrcAmount;
```
Comment (Anyhowclick) [inline contracts/sol6/KyberNetwork.sol:1181]: Will it work if u just `return srcAmount` instead? o.O

**Pull request merged**